addons/mail/models/mail_message.py

```diff
@@ -1171,11 +1171,22 @@ def _notify_message_notification_update(self):
                     continue
                 else:
                     messages |= message
-        updates = [[
-            (self._cr.dbname, 'res.partner', author.id),
-            {'type': 'message_notification_update', 'elements': self.env['mail.message'].concat(*author_messages)._message_notification_format()}
-        ] for author, author_messages in groupby(messages.sorted('author_id'), itemgetter('author_id'))]
-        self.env['bus.bus'].sendmany(updates)
+        notifications = []
+        data_dict = {}
+        for message in messages.sorted('author_id'):
+            for partner in message.notification_ids.mapped('res_partner_id'):
+                log_partner = self.env['res.users.log'].search([('create_uid', 'in', partner.user_ids.ids)], limit=1)
+                if log_partner:
+                    data_dict[partner.id] = message
+            partner = self.env.user.partner_id
+            author = message.author_id
+            if partner.id not in data_dict:
+                data_dict[partner.id] = message
```

Comment:
Each partner in the dict should have a (record)set of messages, not only one. This is how the grouping should be done.